supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala

```diff
@@ -145,33 +147,26 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
     MetadataJobKey(jobDescriptorKey.call.fullyQualifiedName, jobDescriptorKey.index, jobDescriptorKey.attempt)
   }
 
-  override def receive: Receive = LoggingReceive {
+  def jesReceiveBehavior: Receive = LoggingReceive {
     case AbortJobCommand =>
-      serviceRegistryActor ! GetMetadataQueryAction(MetadataQuery.forKey(metadataKey("jobId")))
-    case MetadataLookupResponse(query, events) if query.key.contains("jobId") =>
-      /** Currently this actor can only get a MetadataLookupResponse when requesting a
-        * job ID for abort.  If this actor becomes more complex in the future, then this
-        * should probably be received in an FSM state.
-        */
-      events.headOption foreach { event =>
-        event.value match {
-          case Some(jobIdMetadata) =>
-            val jobId = jobIdMetadata.value
-            jobLogger.info(s"$tag Aborting $jobId")
-            Try(Run(jobId, initializationData.genomics).abort()) match {
-              case Success(_) => jobLogger.info(s"$tag Aborted $jobId")
-              case Failure(ex) => jobLogger.warn(s"$tag Failed to abort $jobId: ${ex.getMessage}")
-            }
-          case None => jobLogger.warn(s"$tag Failed to abort ${jobDescriptor.call.fullyQualifiedName}. Run Id metadata field was empty.")
-        }
+      serviceRegistryActor ! KvGet(ScopedKey(jobDescriptor.descriptor.id, jobDescriptor.key, JesOperationIdKey))
+    case KvPair(scopedKey, operationId) if scopedKey.key == JesOperationIdKey =>
+      operationId match {
+        case Some(id) =>
+          jobLogger.info(s"$tag Aborting $id")
```

Comment:
We use it for both actually.  The above is a somewhat unrelated refactoring of the abort logic to use the KV store and not the metadata service.